woocommerce/woocommerce
# Add business services products to in-app marketplace search results

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR adds a section for Business Services products to the in-app marketplace search results. It also adds the featured business services section added in #47048 to the no-results state of the Search tab.

Changes to the extensions API search endpoint on WooCommerce.com has been done in https://github.com/Automattic/woocommerce.com/pull/20386

Closes https://github.com/Automattic/woocommerce.com/issues/20216 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow steps under the **Prep** heading in https://github.com/woocommerce/woocommerce/pull/47048 to ensure your local WooCommerce dev site can fetch featured products from your local WCCOM.
2. Check out this branch
3. Edit `plugins/woocommerce/includes/admin/class-wc-admin-addons.php` and replace the start of the `fetch_featured()` function with the following:

```php
	public static function fetch_featured() {
		$transient_name = 'wc_addons_featured';
		// Important: WCCOM Extensions API v3.0 is used.
		$url      = 'https://woocommerce.test/wp-json/wccom-extensions/3.0/featured';
		$locale   = get_user_locale();
		$featured = false;
```
(as an alternative to setting `$featured = false` you can also run `wp-env run cli wp transient delete wc_addons_featured` to clear the transient being checked here)

4. In `plugins/woocommerce-admin/client/marketplace/components/constants.ts`, edit the `MARKETPLACE_HOST` const to be `https://woocommerce.test`
5. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
6. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions). Confirm your featured products as set on local WCCOM displays correctly. Switch to other tabs and make sure they still display the correct products/categories. While results are loading, you should see the featured products for the respective product types.
7. In the search box at the top, type a search term that should return one of your Business Services products. Confirm you see a Business Services section in the search results showing that product.
8. Try different search terms that will return combinations of extensions, themes, and business services: where a search term matches two or more product types, results should display a section of results, up to a max of 8, for each type, and a View All button to show only results from that particular type. Where a search term matches only one product type, results should display only products from that type, up to a maximum of 60 products, and the category filters for that type should be visible at the top of the results.
9. Try a search term that won't return any results. Confirm the No Results text appears at the top, followed by the Featured Products sections for "Most popular extensions", "Our favorite themes" and "Services to help your business grow"
10. Test around this issue.

![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/da92670d-9570-4ec8-b1c6-e4af37f3dfed)
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/4e1163ea-ce91-432a-8e0b-fce03bd5b070)
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/85179662-bc5b-4246-b50b-42dcae16b047)
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/5577d8c7-f02f-4f94-9d5b-545f27f84b33)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
